### PR TITLE
feat: add configurable backoffType

### PR DESCRIPTION
**Pull request by findmory**
this adds a new configuration option called `backoffType`

3 options
'exponential' (default)
'static' - uses the retry delay between each retry
'linear' - retry delay * retryAttemptNumber

### CODE REVIEW

**Comment by DanielRuf** [comment]
This would be great. Please merge this.

File: src/index.ts [hunk 1]
```diff
@@ -55,6 +55,11 @@ export interface RetryConfig {
    * When there is no response, the number of retries to attempt. Defaults to 2.
    */
   noResponseRetries?: number;
+
+  /**
+   * Backoff Type; 'linear', 'static' or 'exponential'.
+   */
+  backoffType?: string;
```
Comment (JustinBeckwith) [inline src/index.ts]: Let's use a [string literal type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types) here:

```js
backoffType?: 'linear', 'static', 'exponential';
```

File: src/index.ts [hunk 2]
```diff
@@ -134,7 +139,14 @@ function onError(err: AxiosError) {
   const onBackoffPromise = new Promise(resolve => {
     // Calculate time to wait with exponential backoff.
     // Formula: (2^c - 1 / 2) * 1000
-    const delay = ((Math.pow(2, config.currentRetryAttempt!) - 1) / 2) * 1000;
+    let delay: number;
```
Comment (JustinBeckwith) [inline src/index.ts]: To make this a little more clear, can we set the backoffType?

```js
config.backoffType = config.backoffType || 'expotential';
```

That way we can avoid the checking below :)

File: src/index.ts [hunk 3]
```diff
@@ -134,7 +139,14 @@ function onError(err: AxiosError) {
...
+    if (config.backoffType && config.backoffType === "linear") {
```
Comment (JustinBeckwith) [inline src/index.ts]: Guessing you will need to rebase here after a few other changes I've made, and run `npm run fix` - double quotes are usually turned into single by the formatter

File: src/index.ts (hunk 1 above)
Comment (findmory) [reply src/index.ts]: done

File: src/index.ts (hunk 2 above)
Comment (findmory) [reply src/index.ts]: done

File: src/index.ts (hunk 3 above)
Comment (findmory) [reply src/index.ts]: done

**Pull request merged**